Application/EdFi.Ods.Api.Startup/HttpRouteConfigurations/UserInfoRouteConfiguration.cs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Web.Http;
+using EdFi.Ods.Api.Architecture;
+using EdFi.Ods.Api.Services.Metadata;
+
+namespace EdFi.Ods.Api.Startup.HttpRouteConfigurations
+{
+    public class UserInfoRouteConfiguration : IRouteConfiguration, IOpenApiMetadataRouteConfiguration
+    {
+        public void ConfigureRoutes(HttpConfiguration config, bool useSchoolYear)
+        {
+            config.Routes.MapHttpRoute(
+                name: "UserInfo",
+                routeTemplate: "oauth/userinfo",
```

Comment:
Yeah, I thought we were just putting this at /me?

Reply:
from looking at the spec the /userinfo seems to be more about specific user information (name, email, phone number, address...). it also requires a `sub` claim. I'm not sure this fits very well with the intention of the ticket, what are your thoughts?

Reply:
well that is a good question, we are not fully using the OpenIdConnect spec here, and our model does not fit well. As for the name of the endpoint @vimayya wanted to use userinfo, i put it on the oauth endpoint to match with that domain but i can move it if we all agree.

Reply:
@gmcelhanon suggested using /userinfo. Regarding @tbmeador-dlp comment about the user information (name,email etc.) I thought the example they provided just differed from client credentials flow we use, where client isn't a person. Spec allows returning https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims and any additional claims, however I do see that the format of our content is different, I am not sure if spec allows custom json types like list of ed-orgs.

Reply:
ok will change the endpoint to /me off the root? does that make more sense?

Reply:
another option is vendorInfo? thoughts @vimayya @tbmeador-dlp @blmeyers

Reply:
The spec intends this endpoint to be an introspection endpoint, which matches the use case here, and it allows inclusion of any custom "claims", so I don't see any friction there.

Jim and I discussed this question of where to put this endpoint. Looking at the OpenID Connect and OAuth 2 specs, they describe the endpoints only as `/token` or `/authorize` (etc) -- they do not specify what the URL paths for the endpoints should be. However, as our other OAuth-related endpoints are located under the `/oauth/` path, and taking the view that we should be using a standards-based endpoint for this behavior since it exists, it made sense to me to co-locate rather than put it somewhere else.

We also talked about the OpenID Connect endpoint's use of `sub` (which we don't explicitly provide right now, but could be serviced using the ApiClientId from the Admin database), it seemed pointless to include it in the response.

Perhaps there's a better fit with the OAuth 2.0 `/token_info` endpoint [spec](https://tools.ietf.org/html/rfc7662#section-2.2). That is documented more simply [here](https://www.oauth.com/oauth2-servers/token-introspection-endpoint/).

At the end of the day, `/token_info` is probably a better fit than `/userinfo` since we're still using OAuth 2.0 (not OpenID Connect), but not much else about it would need to change. The only attribute that is required is `active`, and `sub` is defined but OPTIONAL in [the response](https://tools.ietf.org/html/rfc7662#section-2.2).

We are also free to add any additional information to the response that we deem useful:
> Specific implementations MAY extend this structure with their own service-specific response names as top-level members of this JSON object.

As for the location of the endpoint, I'd argue that should be a _sibling_ of the other OAuth-related endpoints that we have which are located under the `/oauth/` path. Why would we want to spread these endpoints out to other locations?

Also, because of [ODS-4033](https://tracker.ed-fi.org/browse/ODS-4033), we should probably be sure to set the `scope` attribute if the token has been requested and issued with scopes applied.

Reply:
I am concerned here. The spec states we would need to use a separate token for introspect and we should also include a post taking a token. Is this what we really want? Is it worth doing this to this spec as a mvp? @vimayya @blmeyers @gmcelhanon @tbmeador-dlp

Reply:
@jamessmckay I am not concerned by that because the spec is written in the context of all of OAuth 2.0 (3 parties involved), not just our `client_credentials` grant flow. In that context, I see this as an indication that it's written to support a superset of our specific use case.

Reply:
@jamessmckay let's go with using OAuth 2.0 /token_info, same bearer token will be sent in the request body as that in the auth header. Optional fields in the response need not be implemented on the first pass. field names should reflect oauth the convention